### ISSUE TITLE
Query editor: visualizations should resize with the available space

### ISSUE THREAD

**Issue by arikfr** [getredash/redash#2455]
Currently visualizations that as widgets resize to their container (like charts or maps) have fixed height in the query editor, which in some cases is bigger than what's currently visible. This creates a poor experiences for users with smaller screens, causing them to constantly have to resize the visualization area to see the visualization even if the allocated space was enough.

**Comment by arikfr** [comment]
@kravets-levko I vaguely remember that we have special handling for this in the CSS code. Do you remember if this is something we can simply remove or there is more to it than that?

**Comment by kravets-levko** [comment]
yes, there is some CSS for this - it is used on query page and on share visualization page. I think, if we want to remove it - we should do it for both pages (and visualizations will fit iframe - we also planned to do it).

**Comment by arikfr** [comment]
Can you point me in the direction so I can play with it a bit? :)

**Comment by kravets-levko** [comment]
NP :+1: 

By default, all visualizations have predefined dimensions, like [this](https://github.com/getredash/redash/blob/master/client/app/assets/less/inc/visualizations/chart.less#L2) or [this](https://github.com/getredash/redash/blob/master/client/app/assets/less/inc/visualizations/map.less#L2). Then for dashboards we override these styles - [here](https://github.com/getredash/redash/blob/master/client/app/assets/less/inc/visualizations/map.less#L2) and maybe other rules may be related too. To make visualizations resizing to it's container, we should move that styles from dashboard CSS to each visualization (not to all - see the sources), and remove that fixed height.
